src/main/scala/strawman/collection/mutable/ArrayDeque.scala

```diff
@@ -0,0 +1,491 @@
+package strawman
+package collection.mutable
+
+import scala._
+import scala.collection.{generic, mutable}
```

Comment:
@julienrf : Design docs would be great! Specifically, in my case, what all should I `extend`? Also, is the internal API more or less "golden"? Last time I looked there were still changes happening and I would rather not devote time porting to a non-frozen internal API...